reference/tools/user-guide.md

```diff
@@ -6,184 +6,270 @@ aliases: ['/docs/dev/how-to/migrate/from-mysql/','/docs/dev/how-to/migrate/incre
+#### TiDB Binlog
 
-To help you understand the relationships between the above tools, here is a brief introduction to TiDB ecosystem tools development roadmap.
+[TiDB Binlog](/reference/tidb-binlog/overview.md) is a system that collects binlog for TiDB clusters and provides tools for near real-time sync and backup.
 
-### TiDB backup and restore
+The following are the basics of TiDB Binlog:
 
-Mydumper and Loader -> BR:
+- Input/Output:
+    - Input: TiDB Cluster
+    - Output: MySQL, TiDB, Kafka or incremental backup files
+- Supported TiDB versions: v2.1 or later
+- Kubernetes support: Yes. See [TiDB Binlog Cluster Operations](/tidb-in-kubernetes/maintain/tidb-binlog.md) and [TiDB Binlog Drainer Configurations in Kubernetes](/tidb-in-kubernetes/reference/configuration/tidb-drainer.md) for details.
+
+#### CDC (Beta, under development, ETA May/June 2020 with TiDB 4.0)
+
+[CDC](/reference/tools/ticdc/overview.md) (Change Data Capture) is a system that collects changelog for key value pairs in TiKV and outputs to downstream systems in transactional order.
+
+- Input/Output: 
+    - Input: TiDB Cluster
+    - Output: MySQL, TiDB, Kafka or incremental backup files
+- Supported TiDB versions: v4.0
+- Kubernetes support: On the development road map, ETA Q2 2020
+
+## Recommended tools for TiDB versions
+
+### Recommended tools for TiDB 3.0 or earlier
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: Use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: Use Mydumper
+- TiDB full data restore:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: Use DM
+- TiDB incremental backup & restore: use TiDB-Binlog
+
+### Recommended tools for TiDB 3.1
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: use BR
+- TiDB full data restore: use BR
+- TiDB incremental backup & restore: use TiDB-Binlog
+
+### Recommended tools for TiDB 4.0
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: use BR
+- TiDB full data restore: use BR
+- TiDB incremental backup & restore: use CDC
+
+## Tools evolution roadmap 
+
+- TiDB Full Data Backup:
+    - Mydumper -> BR
+    - Mydumper -> [dumpling](https://github.com/pingcap/dumpling) (under development, replace Lighting in lightweight scenarios)
+- TiDB Full Data Restore:
+    - Loader -> Lightning -> BR
+- MySQL Data Migration:
+    - Mydumper/Loader + Syncer -> Loader + DM -> Lightning + DM -> DM (in the next step, we will integrate Lightning into DM)
+- TiDB Incremental Data Migration:
+    - TiDB Binlog -> CDC
+
+## Full-path data migration solution for TiDB 3.0, 3.1 and 4.0
```

Comment:
seems this section's most content is duplicate with ` Recommended tools for TiDB versions`

Reply:
It seems that `Full-path data migration solution for TiDB 3.0, 3.1 and 4.0` section gives a more detailed description than `Recommended tools for TiDB versions`. Can we just delete the `Recommended tools for TiDB versions` section? @WangXiangUSTC 

What do you suggest? @GregoryIan @shenli

Reply:
we can merge this PR first and then invite some users to get feedback. We could refine this document based on the feedbacks.

Reply:
Okay, ready for merge now.